ci/unit_tests/functions_delete_package/main_test.py

```diff
@@ -167,7 +167,42 @@ def test_deleteNonexistentPackage(self, useApikey):
         else:
             params.extend(['--cloudfunctions_username', self.username, '--cloudfunctions_password', self.password])
         # Fail
-        self.t_exitCodeAndLogMessage(1, "The resource could not be found. Check your cloudfunctions url and namespace.", [params])
+        self.t_exitCodeAndLogMessage(1, "Package not found. Check your cloudfunctions url and namespace.", [params])
+
+    # TODO: Enable apikey/username+password testing in Nightly builds
+    #@pytest.mark.parametrize('useApikey', [True, False])
+    @pytest.mark.parametrize('useApikey', [True])
+    def test_noPackageProvided(self, useApikey):
+        """Tests if functions_delete_package errors when not providing a package or package name pattern."""
+
+        params = ['-c', os.path.join(self.dataBasePath, 'exampleFunctions.cfg'),
+                '--cloudfunctions_namespace', self.namespace,
+                '--cloudfunctions_url', self.cloudFunctionsUrl]
+
+        if useApikey:
+            params.extend(['--cloudfunctions_apikey', self.apikey])
+        else:
+            params.extend(['--cloudfunctions_username', self.username, '--cloudfunctions_password', self.password])
+        # Fail
+        self.t_exitCodeAndLogMessage(1, "neither 'cloudfunctions_package' nor 'cloudfunctions_package_name_pattern' is defined.", [params])
+
+    # TODO: Enable apikey/username+password testing in Nightly builds
+    #@pytest.mark.parametrize('useApikey', [True, False])
+    @pytest.mark.parametrize('useApikey', [True])
+    def test_noMatchingPackage(self, useApikey):
+        """Tests if functions_delete_package finishes successfully with no matching packages."""
+
+        randomNamePattern = str(uuid.uuid4())
+        params = ['-c', os.path.join(self.dataBasePath, 'exampleFunctions.cfg'),
+                '--cloudfunctions_package_name_pattern', randomNamePattern, '--cloudfunctions_namespace', self.namespace,
+                '--cloudfunctions_url', self.cloudFunctionsUrl]
+
+        if useApikey:
+            params.extend(['--cloudfunctions_apikey', self.apikey])
+        else:
+            params.extend(['--cloudfunctions_username', self.username, '--cloudfunctions_password', self.password])
+
+        self.t_noExceptionAndLogMessage("No matching packages to delete.", [params])
```

Comment:
There is no package at the beginning, right? It would be better to first create some package, then try to delete non-matching, and then test if package is still present.

Reply:
There are several other packages as it's a testing namespace shared between multiple people AND our CI.

Reply:
Done in a65a31d.